src/platforms/common/stm32/serialno.c

```diff
@@ -64,3 +68,44 @@ void read_serial_number(void)
 #endif
 	serial_no[DFU_SERIAL_LENGTH - 1] = '\0';
 }
+
+/** @brief Convert an integer value into printable base-N representation.
```

Comment:
Good idea to document what's what, though for consistency with the rest of the code, please let's not use (quasi-?)doxygen syntax. Likewise, `/*` on its own line.

Reply:
This is roughly the style of libopencm3 on which this firmware relies on: `/** @brief` and multiline C++ down. I thought to document somehow that this function behaves slightly differently than `utoa()` would. Most of the block can be dropped except `@param[in] str` and return value description.

Reply:
The suggestion is more to switch the style over to something like you'd find elsewhere in the code base, eg:
```c
/*
 * Converts an unsigned integer value into an equivilent base-N ASCII printable
 * representation. It takes as arguments (in order): the value to convert,
 * a pointer to a buffer large enough to take the result, and the base
 * to perform the conversion in between 2 and 36 (inclusive).
 * The result of this function is NULL on error (unsupported base), or a pointer
 * into the input buffer advanced by the amount written (not including the
 * nul terminator guaranteed by this function)
 */
 ```

Reply:
Reflowed text per review item, throwing away Doxygen tags intended for IDE and HTML docs. Codespell says there's a stray "desig" in libopencm3 header filename, but nothing more.